### PR TITLE
Java @Override Flag Support

### PR DESCRIPTION
**Pull request by TanayParikh**
- Added support for `@Override` flag. Flag permits for better error handling.
> When overriding a method, you might want to use the `@Override` annotation that instructs the compiler that you intend to override a method in the superclass. If, for some reason, the compiler detects that the method does not exist in one of the superclasses, then it will generate an error. For more information on `@Override`, see Annotations.
> https://docs.oracle.com/javase/tutorial/java/IandI/override.html

- Fixed `documentation` typo in nanos

### CODE REVIEW

File: src/foam/java/Method.js [hunk 1]
```diff
@@ -30,12 +30,22 @@ foam.CLASS({
...
     { class: 'StringArray', name: 'throws' },
     { class: 'foam.java.CodeProperty', name: 'body' }
   ],
 
   methods: [
     function outputJava(o) {
+      if (this.override) {
+        o.indent();
+        o.out("@Override\n");
+      }
```
Comment (kirkeaton) [inline src/foam/java/Method.js:47]: Are you missing an `o.indent();` here or does InterfaceMethod.js just have an extra one?

**Review by kirkeaton — APPROVED** [review]

File: src/foam/java/Method.js (hunk 1 above)
Comment (TanayParikh) [reply src/foam/java/Method.js:47]: Tested again, extra indent is needed for proper formatting. Nice catch though :)

**Comment by kgrgreer** [comment]
I disagree with this approach. Would be better to validate at the Model level rather than leaving for the Java compiler to detect. This method only works for Java, not JS or Swift, and catches the error much later in the pipeline than if we just did it in FOAM (which in the future, could be flagged right in the IDE).

**Comment by TanayParikh** [comment]
Thanks for the clarification, closing for now.

**Pull request closed without merging**